### PR TITLE
feat: create bank smart contract

### PR DESCRIPTION
**Pull request by farrellh1**
# Description

create banking smart contract
for details of the functionality check the issue below

Fixes # (#124)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

* **Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)

### CODE REVIEW

**Comment by hermannleboss** [comment]
@farrellh1 can you create all the bank account.

**Comment by farrellh1** [comment]
idk why the ci cd not running

**Comment by hermannleboss** [comment]
> idk why the ci cd not running

The CI CD run depending on what file is updated

**Comment by hermannleboss** [comment]
For the moment the CI CD for the contract run only on PR on develop and main branch

File: contract/test/Bank.spec.ts
[the remark this answers is not in the document]
Comment (hermannleboss) [reply contract/test/Bank.spec.ts:1]: Can you look at this for the test structure. 
https://dev.to/walmyrlimaesilv/using-the-keywords-given-when-then-with-cypress-but-without-cucumber-118p

It's about cypress but it work with most test framework.

**Review by hermannleboss — CHANGES_REQUESTED** [review]

File: contract/test/Bank.spec.ts
[the remark this answers is not in the document]
Comment (farrellh1) [reply contract/test/Bank.spec.ts:1]: i see, it's like a user stories, right?

[the remark this answers is not in the document]
Comment (farrellh1) [reply contract/test/Bank.spec.ts:1]: i was following how Uniswap testing their smart contract

**Comment by farrellh1** [comment]
i applied your suggestion @hermannleboss , but it reduce the code coverage

File: contract/test/Bank.spec.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        })
+      })
+    })
+  })
+
+  // As a company owner of a digital service provider.
+  // I want to direct money from my online operations to my team’s crypto “bank” account on the CNC portal.
+  // When I deposit into my team's bank contract
+  // Then I can see funds in the team's bank contract
+
```
Comment (hermannleboss) [inline contract/test/Bank.spec.ts:61]: It's from there i remark you are using the user Stories i create for the CNC portal to make the test. 
Let me tell you that there is a difference between the userStories in the context of the CNC and in the context of the Bank Smart contract

**Review by hermannleboss — CHANGES_REQUESTED** [review]
Take the user Stories in the context of the bank.

**Comment by hermannleboss** [comment]
> > r Stories of the Bank Smart Contract
> 
> should i revise all the testing to be like your suggestion instead @hermannleboss ? or just adjust some of them to be like you mentioned here?

You should refactor you test to the new useStories. You adjust what i wrote.

**Comment by farrellh1** [comment]
applied @hermannleboss , thanks

**Review by hermannleboss — APPROVED** [review]
LGTM

**Pull request merged**